mesonbuild/msetup.py

```diff
@@ -111,7 +116,11 @@ def has_build_file(self, dirname: str) -> bool:
         fname = os.path.join(dirname, environment.build_filename)
         return os.path.exists(fname)
 
-    def validate_core_dirs(self, dir1: str, dir2: str) -> T.Tuple[str, str]:
+    def validate_core_dirs(self, dir1: str, dir2: str, builddir: str) -> T.Tuple[str, str]:
+        if builddir is not None:
+            if dir2 is not None:
+                raise MesonException('builddir defined with -C argument and positional argument')
```

Comment:
Simpler: "when -C is specified, only one positional argument is specified".

Reply:
ok, updated.